hazelcast/src/test/java/com/hazelcast/query/impl/predicates/BetweenVisitorTest.java

```diff
@@ -0,0 +1,208 @@
+package com.hazelcast.query.impl.predicates;
+
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.FalsePredicate;
+import com.hazelcast.query.impl.Index;
+import com.hazelcast.query.impl.Indexes;
+import com.hazelcast.query.impl.TypeConverters;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.query.Predicates.*;
+import static com.hazelcast.query.impl.TypeConverters.INTEGER_CONVERTER;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class BetweenVisitorTest {
+
```

Comment:
I mean expressions that couldn't be flattened.
I know that this visitor expects flattened input, but there are certainly cases where the input to this visitor won't be flat, e.g. 
`(age >= 18 or ((height >= 170 and weight<55) || (height >= 180 and weight<60)) or age <= 28)`